libkbfs/tlf_edit_history.go

```diff
@@ -493,22 +503,40 @@ func (teh *TlfEditHistory) updateHistory(ctx context.Context,
 				"op for original ptr %v", original)
 		}
 		oldPath := oldParentChain.ops[0].getFinalPath().
-			ChildPathNoPtr(ri.oldName).String()
+			ChildPathNoPtr(ri.oldName)
 
 		newParentChain, ok := chains.byOriginal[ri.originalNewParent]
 		if !ok || len(newParentChain.ops) == 0 {
 			teh.log.CDebugf(ctx, "Couldn't find new parent to a rename "+
 				"op for original ptr %v", original)
 		}
-		newPath := newParentChain.ops[0].getFinalPath().
-			ChildPathNoPtr(ri.newName).String()
+
+		var renameCreate *createOp
+		for _, op := range newParentChain.ops {
+			if cop, ok := op.(*createOp); ok && cop.renamed &&
+				cop.NewName == ri.newName {
+				renameCreate = cop
+				break
+			}
+		}
+		if renameCreate == nil {
+			return fmt.Errorf("Couldn't find the create op for the %s->%s "+
+				"rename", ri.oldName, ri.newName)
+		}
+
+		newPath := renameCreate.getFinalPath().ChildPathNoPtr(ri.newName)
 
 		// Ignore any previous rmOps.
-		delete(removed, oldPath)
+		delete(removed, oldPath.String())
+		delete(removeNotifications, oldPath.String())
 		// If a file was overwritten, ignore all the old edits.
-		removed[newPath] = true
+		removed[newPath.String()] = true
```

Comment:
Then `TlfWriterEdits.updateOldEdits()` would ignore it and only the rename would take place.